azurerm_mssql_managed_database: yearly_retention defaults to PT0S causing week_of_year validation error and state drift in long_term_retention_policy

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.12.2

### AzureRM Provider Version

4.69.0

### Affected Resource(s)/Data Source(s)

azurerm_mssql_managed_database

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
   name     = "rg-sqlmi-example"
   location = "switzerlandnorth"
 }
 
 resource "azurerm_mssql_managed_instance" "example" {
   name                = "example-sqlmi"
   resource_group_name = azurerm_resource_group.example.name
   location            = azurerm_resource_group.example.location
 
   license_type       = "BasePrice"
   sku_name           = "GP_Gen5"
   vcores             = 4
   storage_size_in_gb = 32
 
   administrator_login          = "sqladmin"
   administrator_login_password = "Pa$$w0rd1234!"
 }
 
 resource "azurerm_mssql_managed_database" "example" {
   name                      = "exampledb"
   managed_instance_id       = azurerm_mssql_managed_instance.example.id
   short_term_retention_days = 7
 
   long_term_retention_policy {
     weekly_retention = "P7D"
   }
 }
```

### Debug Output/Panic Output

```shell
2026-04-23T15:08:10.274+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: {"name":"98658096-1424-458a-a5d7-b89f5f883853","status":"Succeeded","startTime":"2026-04-23T13:07:39.767Z"}
2026-04-23T15:08:10.275+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: [DEBUG] AzureRM Request: 
2026-04-23T15:08:10.275+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: PUT /subscriptions/***Redacted***/resourceGroups/rg-sqlmi-example/providers/Microsoft.Sql/managedInstances/example-sqlmi/databases/exampledb/backupLongTermRetentionPolicies/default?api-version=2023-08-01-preview HTTP/1.1
2026-04-23T15:08:10.275+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: Host: management.azure.com
2026-04-23T15:08:10.275+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: User-Agent: HashiCorp/go-azure-sdk (Go-http-Client/1.1 managedinstancelongtermretentionpolicies/2023-08-01-preview) HashiCorp Terraform/1.10.4 (+https://www.terraform.io) terraform-provider-azurerm/4.69.0 pid-222c6c49-1b0a-5959-a213-6608f9eb8820
2026-04-23T15:08:10.275+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: Content-Length: 98
2026-04-23T15:08:10.275+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: Content-Type: application/json; charset=utf-8
2026-04-23T15:08:10.275+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: X-Ms-Correlation-Request-Id: 41ec7548-9c42-7e5a-a1d7-10240f41aeaa
2026-04-23T15:08:10.275+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: Accept-Encoding: gzip
2026-04-23T15:08:10.275+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5
2026-04-23T15:08:10.275+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: {"properties":{"monthlyRetention":"","weekOfYear":0,"weeklyRetention":"P7D","yearlyRetention":""}}
2026-04-23T15:08:10.275+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: [DEBUG] PUT https://management.azure.com/subscriptions/***Redacted***/resourceGroups/rg-sqlmi-example/providers/Microsoft.Sql/managedInstances/example-sqlmi/databases/exampledb/backupLongTermRetentionPolicies/default?api-version=2023-08-01-preview
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: [DEBUG] AzureRM Response for https://management.azure.com/subscriptions/***Redacted***/resourceGroups/rg-sqlmi-example/providers/Microsoft.Sql/managedInstances/example-sqlmi/databases/exampledb/backupLongTermRetentionPolicies/default?api-version=2023-08-01-preview: 
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: HTTP/2.0 400 Bad Request
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: Content-Length: 151
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: Cache-Control: no-cache
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: Content-Type: application/json; charset=utf-8
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: Date: Thu, 23 Apr 2026 13:08:09 GMT
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: Expires: -1
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: Pragma: no-cache
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: Strict-Transport-Security: max-age=31536000; includeSubDomains
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: X-Cache: CONFIG_NOCACHE
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: X-Content-Type-Options: nosniff
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: X-Ms-Correlation-Request-Id: 41ec7548-9c42-7e5a-a1d7-10240f41aeaa
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: X-Ms-Operation-Identifier: appId=***Redacted***,tenantId=***Redacted***,objectId=***Redacted***/switzerlandnorth/ff76a21e-f95b-4eea-9a06-00d833eaebc2
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: X-Ms-Ratelimit-Remaining-Subscription-Global-Writes: 2999
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: X-Ms-Ratelimit-Remaining-Subscription-Writes: 199
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: X-Ms-Request-Id: d5a5d8f6-b5a9-4b81-8426-bdb67e81fd06
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: X-Ms-Routing-Request-Id: SWITZERLANDNORTH:20260423T130810Z:5e5b869a-5b64-4ad7-b659-c3c3c0798d96
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: X-Msedge-Ref: Ref A: 1431C718B6184712AC22AD0369BF6FFB Ref B: GVA201070811025 Ref C: 2026-04-23T13:08:10Z
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5
2026-04-23T15:08:10.481+0200 [DEBUG] provider.terraform-provider-azurerm_v4.69.0_x5: {"error":{"code":"LongTermRetentionMissingWeekOfYear","message":"WeekOfYear is required to be set between 1 and 52 in order to set yearly retention."}}
2026-04-23T15:08:10.481+0200 [ERROR] provider.terraform-provider-azurerm_v4.69.0_x5: Response contains error diagnostic: tf_provider_addr=registry.terraform.io/hashicorp/azurerm diagnostic_severity=ERROR
  diagnostic_detail=
  | setting Long Term Retention Policies for Managed Instance Database (Subscription: "***Redacted***"
  | Resource Group Name: "rg-sqlmi-example"
  | Managed Instance Name: "example-sqlmi"
  | Database Name: "exampledb"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: LongTermRetentionMissingWeekOfYear: WeekOfYear is required to be set between 1 and 52 in order to set yearly retention.
  
  diagnostic_summary=
  | setting Long Term Retention Policies for Managed Instance Database (Subscription: "***Redacted***"
  | Resource Group Name: "rg-sqlmi-example"
  | Managed Instance Name: "example-sqlmi"
  | Database Name: "exampledb"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: LongTermRetentionMissingWeekOfYear: WeekOfYear is required to be set between 1 and 52 in order to set yearly retention.
   tf_proto_version=5.10 tf_req_id=14f67a48-f955-f759-f4d7-cfdbc0fc997b tf_resource_type=azurerm_mssql_managed_database @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/diag/diagnostics.go:58 @module=sdk.proto tf_rpc=ApplyResourceChange timestamp="2026-04-23T15:08:10.481+0200"
{"@level":"info","@message":"azurerm_mssql_managed_database.example: Creation errored after 31s","@module":"terraform.ui","@timestamp":"2026-04-23T15:08:10.482363+02:00","hook":{"resource":{"addr":"azurerm_mssql_managed_database.example","module":"","resource":"azurerm_mssql_managed_database.example","implied_provider":"azurerm","resource_type":"azurerm_mssql_managed_database","resource_name":"example","resource_key":null},"action":"create","elapsed_seconds":31},"type":"apply_errored"}
2026-04-23T15:08:10.482+0200 [DEBUG] State storage *cloud.State declined to persist a state snapshot
2026-04-23T15:08:10.482+0200 [ERROR] vertex "azurerm_mssql_managed_database.example" error: setting Long Term Retention Policies for Managed Instance Database (Subscription: "***Redacted***"
Resource Group Name: "rg-sqlmi-example"
Managed Instance Name: "example-sqlmi"
Database Name: "exampledb"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: LongTermRetentionMissingWeekOfYear: WeekOfYear is required to be set between 1 and 52 in order to set yearly retention.
```

### Expected Behaviour

 When yearly_retention is omitted (or explicitly set to "PT0S", meaning no yearly backup), the provider should:

   1. Not require week_of_year to be set, since yearly retention is disabled.
   2. Not send a value for week_of_year to the Azure API (or send 0/null without triggering a validation error).
   3. Not produce state drift on subsequent terraform plan runs because the Azure API returns week_of_year = 0 when yearly retention is disabled.

### Actual Behaviour

 When long_term_retention_policy is configured with only weekly_retention = "P7D" (omitting yearly_retention and week_of_year):

   1. terraform plan succeeds without any warnings.
   2. terraform apply fails with the following error:

   Error: week_of_year must be a value between 1 and 52

  This happens because yearly_retention defaults to "PT0S" internally, which causes Azure to set week_of_year = 0 a value that fails Azure's own validation (valid range: 1–52).

  Even if a workaround is applied (e.g. explicitly setting yearly_retention = "PT0S"), subsequent terraform plan runs show perpetual drift because the Azure API returns week_of_year = 0 while Terraform
  expects no value (or a different representation).


### Steps to Reproduce

   1. Define an azurerm_mssql_managed_database resource with a long_term_retention_policy block that sets only weekly_retention = "P7D", omitting yearly_retention and week_of_year.
   2. Run terraform init
   3. Run terraform plan — plan succeeds with no errors.
   4. Run terraform apply — apply fails with: week_of_year must be a value between 1 and 52

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/mssqlmanagedinstance/mssql_managed_database_data_source.go
+++ b/internal/services/mssqlmanagedinstance/mssql_managed_database_data_source.go
@@ -12,6 +12,7 @@ import (
 	"github.com/hashicorp/go-azure-helpers/lang/response"
 	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/features"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssqlmanagedinstance/validate"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
@@ -61,15 +62,17 @@ func (d MsSqlManagedDatabaseDataSource) Arguments() map[string]*pluginsdk.Schema
 }
 
 func (d MsSqlManagedDatabaseDataSource) Attributes() map[string]*pluginsdk.Schema {
-	return map[string]*pluginsdk.Schema{
+	attrs := map[string]*pluginsdk.Schema{
 		"managed_instance_name": {
 			Type:     pluginsdk.TypeString,
 			Computed: true,
 		},
+
 		"resource_group_name": {
 			Type:     pluginsdk.TypeString,
 			Computed: true,
 		},
+
 		"long_term_retention_policy": {
 			Type:     pluginsdk.TypeList,
 			Computed: true,
@@ -94,18 +97,15 @@ func (d MsSqlManagedDatabaseDataSource) Attributes() map[string]*pluginsdk.Schem
 						Type:     pluginsdk.TypeInt,
 						Computed: true,
 					},
-
-					"immutable_backups_enabled": {
-						Type:     pluginsdk.TypeBool,
-						Computed: true,
-					},
 				},
 			},
 		},
+
 		"short_term_retention_days": {
 			Type:     pluginsdk.TypeInt,
 			Computed: true,
 		},
+
 		"point_in_time_restore": {
 			Type:     schema.TypeList,
 			Computed: true,
@@ -123,6 +123,16 @@ func (d MsSqlManagedDatabaseDataSource) Attributes() map[string]*pluginsdk.Schem
 			},
 		},
 	}
+
+	if !features.FivePointOh() {
+		attrs["long_term_retention_policy"].Elem.(*pluginsdk.Resource).Schema["immutable_backups_enabled"] = &pluginsdk.Schema{
+			Type:       pluginsdk.TypeBool,
+			Computed:   true,
+			Deprecated: "The `long_term_retention_policy.immutable_backups_enabled` property has been deprecated and will be removed in v5.0 of the AzureRM provider. This property is non-functional and was mistakenly exposed in the data source.",
+		}
+	}
+
+	return attrs
 }
 
 func (d MsSqlManagedDatabaseDataSource) Read() sdk.ResourceFunc {

--- a/internal/services/mssqlmanagedinstance/mssql_managed_database_resource.go
+++ b/internal/services/mssqlmanagedinstance/mssql_managed_database_resource.go
@@ -20,7 +20,6 @@ import (
 	helperValidate "github.com/hashicorp/terraform-provider-azurerm/helpers/validate"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/features"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssql/helper"
 	miParse "github.com/hashicorp/terraform-provider-azurerm/internal/services/mssqlmanagedinstance/parse"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssqlmanagedinstance/validate"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
@@ -69,6 +68,11 @@ func (r MsSqlManagedDatabaseResource) IDValidationFunc() pluginsdk.SchemaValidat
 }
 
 func (r MsSqlManagedDatabaseResource) Arguments() map[string]*pluginsdk.Schema {
+	atLeastOneOf := []string{
+		"long_term_retention_policy.0.weekly_retention", "long_term_retention_policy.0.monthly_retention",
+		"long_term_retention_policy.0.yearly_retention", "long_term_retention_policy.0.week_of_year",
+	}
+
 	resource := map[string]*pluginsdk.Schema{
 		"name": {
 			Type:         pluginsdk.TypeString,
@@ -84,7 +88,51 @@ func (r MsSqlManagedDatabaseResource) Arguments() map[string]*pluginsdk.Schema {
 			ValidateFunc: validate.ManagedInstanceID,
 		},
 
-		"long_term_retention_policy": helper.LongTermRetentionPolicySchema(),
+		"long_term_retention_policy": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Computed: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					// WeeklyRetention - The weekly retention policy for an LTR backup in an ISO 8601 format.
+					"weekly_retention": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						Default:      "PT0S",
+						ValidateFunc: helperValidate.ISO8601Duration,
+						AtLeastOneOf: atLeastOneOf,
+					},
+
+					// MonthlyRetention - The monthly retention policy for an LTR backup in an ISO 8601 format.
+					"monthly_retention": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						Default:      "PT0S",
+						ValidateFunc: helperValidate.ISO8601Duration,
+						AtLeastOneOf: atLeastOneOf,
+					},
+
+					// YearlyRetention - The yearly retention policy for an LTR backup in an ISO 8601 format.
+					"yearly_retention": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						Default:      "PT0S",
+						ValidateFunc: helperValidate.ISO8601Duration,
+						AtLeastOneOf: atLeastOneOf,
+					},
+
+					// WeekOfYear - The week of year to take the yearly backup.
+					"week_of_year": {
+						Type:         pluginsdk.TypeInt,
+						Optional:     true,
+						Computed:     true,
+						ValidateFunc: validation.IntBetween(1, 52),
+						AtLeastOneOf: atLeastOneOf,
+					},
+				},
+			},
+		},
 
 		"short_term_retention_days": {
 			Type:         pluginsdk.TypeInt,
@@ -121,10 +169,6 @@ func (r MsSqlManagedDatabaseResource) Arguments() map[string]*pluginsdk.Schema {
 	}
 
 	if !features.FivePointOh() {
-		atLeastOneOf := []string{
-			"long_term_retention_policy.0.weekly_retention", "long_term_retention_policy.0.monthly_retention",
-			"long_term_retention_policy.0.yearly_retention", "long_term_retention_policy.0.week_of_year",
-		}
 		resource["long_term_retention_policy"] = &pluginsdk.Schema{
 			Type:     pluginsdk.TypeList,
 			Optional: true,
@@ -169,9 +213,10 @@ func (r MsSqlManagedDatabaseResource) Arguments() map[string]*pluginsdk.Schema {
 					},
 
 					"immutable_backups_enabled": {
-						Type:     pluginsdk.TypeBool,
-						Optional: true,
-						Default:  false,
+						Type:       pluginsdk.TypeBool,
+						Optional:   true,
+						Default:    false,
+						Deprecated: "The `long_term_retention_policy.immutable_backups_enabled` property has been deprecated and will be removed in v5.0 of the AzureRM provider. This property is non-functional and was mistakenly exposed in the resource.",
 					},
 				},
 			},
@@ -249,11 +294,10 @@ func (r MsSqlManagedDatabaseResource) Create() sdk.ResourceFunc {
 
 			if len(model.LongTermRetentionPolicy) > 0 {
 				longTermRetentionPolicy := managedinstancelongtermretentionpolicies.ManagedInstanceLongTermRetentionPolicy{
-					Properties: pointer.To(expandLongTermRetentionPolicy(model.LongTermRetentionPolicy)),
+					Properties: expandLongTermRetentionPolicy(model.LongTermRetentionPolicy),
 				}
 
-				err := longTermRetentionClient.CreateOrUpdateThenPoll(ctx, id, longTermRetentionPolicy)
-				if err != nil {
+				if err := longTermRetentionClient.CreateOrUpdateThenPoll(ctx, id, longTermRetentionPolicy); err != nil {
 					return fmt.Errorf("setting Long Term Retention Policies for %s: %+v", id, err)
 				}
 			}
@@ -319,11 +363,10 @@ func (r MsSqlManagedDatabaseResource) Update() sdk.ResourceFunc {
 
 			if d.HasChange("long_term_retention_policy") {
 				longTermRetentionPolicy := managedinstancelongtermretentionpolicies.ManagedInstanceLongTermRetentionPolicy{
-					Properties: pointer.To(expandLongTermRetentionPolicy(model.LongTermRetentionPolicy)),
+					Properties: expandLongTermRetentionPolicy(model.LongTermRetentionPolicy),
 				}
 
-				err := longTermRetentionClient.CreateOrUpdateThenPoll(ctx, id, longTermRetentionPolicy)
-				if err != nil {
+				if err := longTermRetentionClient.CreateOrUpdateThenPoll(ctx, id, longTermRetentionPolicy); err != nil {
 					return fmt.Errorf("updating Long Term Retention Policies for %s: %+v", id, err)
 				}
 			}
@@ -428,17 +471,25 @@ func (r MsSqlManagedDatabaseResource) Delete() sdk.ResourceFunc {
 	}
 }
 
-func expandLongTermRetentionPolicy(ltrPolicy []LongTermRetentionPolicy) managedinstancelongtermretentionpolicies.ManagedInstanceLongTermRetentionPolicyProperties {
+func expandLongTermRetentionPolicy(ltrPolicy []LongTermRetentionPolicy) *managedinstancelongtermretentionpolicies.ManagedInstanceLongTermRetentionPolicyProperties {
 	if len(ltrPolicy) == 0 {
-		return managedinstancelongtermretentionpolicies.ManagedInstanceLongTermRetentionPolicyProperties{}
+		return &managedinstancelongtermretentionpolicies.ManagedInstanceLongTermRetentionPolicyProperties{}
 	}
 
-	return managedinstancelongtermretentionpolicies.ManagedInstanceLongTermRetentionPolicyProperties{
+	// The API errors if `weekOfYear` is not at least 1 when `yearlyRetention` is provided.
+	weekOfYear := int64(1)
+	if ltrPolicy[0].WeekOfYear != 0 {
+		weekOfYear = ltrPolicy[0].WeekOfYear
+	}
+
+	result := &managedinstancelongtermretentionpolicies.ManagedInstanceLongTermRetentionPolicyProperties{
 		WeeklyRetention:  &ltrPolicy[0].WeeklyRetention,
 		MonthlyRetention: &ltrPolicy[0].MonthlyRetention,
 		YearlyRetention:  &ltrPolicy[0].YearlyRetention,
-		WeekOfYear:       pointer.To(ltrPolicy[0].WeekOfYear),
+		WeekOfYear:       &weekOfYear,
 	}
+
+	return result
 }
 
 func flattenLongTermRetentionPolicy(ltrPolicy managedinstancelongtermretentionpolicies.ManagedInstanceLongTermRetentionPolicyProperties) []LongTermRetentionPolicy {

--- a/internal/services/mssqlmanagedinstance/mssql_managed_database_resource.go
+++ b/internal/services/mssqlmanagedinstance/mssql_managed_database_resource.go
@@ -127,7 +127,7 @@ func (r MsSqlManagedDatabaseResource) Arguments() map[string]*pluginsdk.Schema {
 						Type:         pluginsdk.TypeInt,
 						Optional:     true,
 						Computed:     true,
-						ValidateFunc: validation.IntBetween(1, 52),
+						ValidateFunc: validation.IntBetween(0, 52),
 						AtLeastOneOf: atLeastOneOf,
 					},
 				},

--- a/internal/services/mssqlmanagedinstance/mssql_managed_database_resource.go
+++ b/internal/services/mssqlmanagedinstance/mssql_managed_database_resource.go
@@ -420,13 +420,16 @@ func (r MsSqlManagedDatabaseResource) Read() sdk.ResourceFunc {
 				ManagedInstanceId: managedInstanceId.ID(),
 			}
 
-			ltrResp, err := longTermRetentionClient.Get(ctx, *id)
-			if err != nil {
-				return fmt.Errorf("retrieving Long Term Retention Policy for  %s: %v", *id, err)
-			}
+			// If the database is in a `Stopped` state, attempting to retrieve the long term retention policy results in a 400.
+			if result.Model != nil && result.Model.Properties != nil && pointer.From(result.Model.Properties.Status) != manageddatabases.ManagedDatabaseStatusStopped {
+				ltrResp, err := longTermRetentionClient.Get(ctx, *id)
+				if err != nil {
+					return fmt.Errorf("retrieving Long Term Retention Policy for %s: %v", *id, err)
+				}
 
-			if ltrResp.Model != nil && ltrResp.Model.Properties != nil {
-				model.LongTermRetentionPolicy = flattenLongTermRetentionPolicy(*ltrResp.Model.Properties)
+				if ltrResp.Model != nil && ltrResp.Model.Properties != nil {
+					model.LongTermRetentionPolicy = flattenLongTermRetentionPolicy(*ltrResp.Model.Properties)
+				}
 			}
 
 			shortTermRetentionResp, err := shortTermRetentionClient.Get(ctx, *id)

--- a/internal/services/mssqlmanagedinstance/mssql_managed_database_resource.go
+++ b/internal/services/mssqlmanagedinstance/mssql_managed_database_resource.go
@@ -430,6 +430,9 @@ func (r MsSqlManagedDatabaseResource) Read() sdk.ResourceFunc {
 				if ltrResp.Model != nil && ltrResp.Model.Properties != nil {
 					model.LongTermRetentionPolicy = flattenLongTermRetentionPolicy(*ltrResp.Model.Properties)
 				}
+			} else {
+				// Preserve state while database is `Stopped`
+				model.LongTermRetentionPolicy = state.LongTermRetentionPolicy
 			}
 
 			shortTermRetentionResp, err := shortTermRetentionClient.Get(ctx, *id)

--- a/internal/services/mssqlmanagedinstance/mssql_managed_database_resource_test.go
+++ b/internal/services/mssqlmanagedinstance/mssql_managed_database_resource_test.go
@@ -89,13 +89,61 @@ func TestAccMsSqlManagedDatabase_withRetentionPolicies(t *testing.T) {
 				check.That(data.ResourceName).ExistsInAzure(r),
 			),
 		},
-		data.ImportStep(""),
+		data.ImportStep(),
 		{
 			Config: r.withRetentionPoliciesUpdated(data),
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
 			),
 		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccMsSqlManagedDatabase_withPartialRetentionPolicies(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_mssql_managed_database", "test")
+	r := MsSqlManagedDatabase{}
+
+	weekly := `weekly_retention = "P10D"`
+	monthly := `monthly_retention = "P10D"`
+	yearly := `yearly_retention = "P2Y"`
+	yearlyWithWeekOfYear := `yearly_retention = "P3Y"` + "\nweek_of_year = 42"
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.withPartialRetentionPolicies(data, weekly),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.withPartialRetentionPolicies(data, monthly),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.withPartialRetentionPolicies(data, yearly),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config: r.withPartialRetentionPolicies(data, yearlyWithWeekOfYear),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.withPartialRetentionPolicies(data, weekly),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
 	})
 }
 
@@ -199,6 +247,21 @@ resource "azurerm_mssql_managed_database" "test" {
 
 }
 `, MsSqlManagedInstanceResource{}.basic(data), data.RandomInteger)
+}
+
+func (r MsSqlManagedDatabase) withPartialRetentionPolicies(data acceptance.TestData, retention string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_mssql_managed_database" "test" {
+  managed_instance_id = azurerm_mssql_managed_instance.test.id
+  name                = "acctest-%[2]d"
+
+  long_term_retention_policy {
+    %[3]s
+  }
+}
+`, MsSqlManagedInstanceResource{}.basic(data), data.RandomInteger, retention)
 }
 
 func (r MsSqlManagedDatabase) pointInTimeRestore(data acceptance.TestData, restorePointInTime string) string {

--- a/website/docs/5.0-upgrade-guide.html.markdown
+++ b/website/docs/5.0-upgrade-guide.html.markdown
@@ -489,11 +489,12 @@ Please follow the format in the example below for listing breaking changes in re
 
 ### `azurerm_mssql_database`
 
-* The properties `weekly_retention`, `monthly_retention` and `yearly_retention` now default to `PT0S`.
+* The `long_term_retention_policy.weekly_retention`, `long_term_retention_policy.monthly_retention` and `long_term_retention_policy.yearly_retention` properties now default to `PT0S`.
 
 ### `azurerm_mssql_managed_database`
 
-* The properties `weekly_retention`, `monthly_retention` and `yearly_retention` now default to `PT0S`.
+* The deprecated `long_term_retention_policy.immutable_backups_enabled` property has been removed.
+* The `long_term_retention_policy.weekly_retention`, `long_term_retention_policy.monthly_retention` and `long_term_retention_policy.yearly_retention` properties now default to `PT0S`.
 
 ### `azurerm_mssql_managed_instance`
 
@@ -701,6 +702,10 @@ Please follow the format in the example below for listing breaking changes in da
 ### `azurerm_logic_app_standard`
 
 * The deprecated `site_config.public_network_access_enabled` property has been removed and superseded by the `public_network_access` property.
+
+### `azurerm_mssql_managed_database`
+
+* The deprecated `long_term_retention_policy.immutable_backups_enabled` property has been removed.
 
 ### `azurerm_nginx_configuration`
 

--- a/website/docs/d/mssql_managed_database.html.markdown
+++ b/website/docs/d/mssql_managed_database.html.markdown
@@ -48,8 +48,6 @@ In addition to the Arguments listed above - the following Attributes are exporte
 
 A `long_term_retention_policy` block exports the following:
 
-* `immutable_backups_enabled` - Specifies if the backups are immutable.
-
 * `monthly_retention` - The monthly retention policy for an LTR backup in an ISO 8601 format.
 
 * `week_of_year` - The week of year to take the yearly backup.

--- a/website/docs/r/mssql_managed_database.html.markdown
+++ b/website/docs/r/mssql_managed_database.html.markdown
@@ -80,11 +80,10 @@ The following arguments are supported:
 
 A `long_term_retention_policy` block supports the following:
 
-* `weekly_retention` - (Optional) The weekly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 to 520 weeks. e.g. `P1Y`, `P1M`, `P1W` or `P7D`. Defaults to `PT0S`.
-* `monthly_retention` - (Optional) The monthly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 to 120 months. e.g. `P1Y`, `P1M`, `P4W` or `P30D`. Defaults to `PT0S`.
-* `yearly_retention` - (Optional) The yearly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 to 10 years. e.g. `P1Y`, `P12M`, `P52W` or `P365D`. Defaults to `PT0S`.
+* `weekly_retention` - (Optional) The weekly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 and 520 weeks. e.g. `P1Y`, `P1M`, `P1W` or `P7D`. Defaults to `PT0S`.
+* `monthly_retention` - (Optional) The monthly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 and 120 months. e.g. `P1Y`, `P1M`, `P4W` or `P30D`. Defaults to `PT0S`.
+* `yearly_retention` - (Optional) The yearly retention policy for an LTR backup in an ISO 8601 format. Valid value is between 1 and 10 years. e.g. `P1Y`, `P12M`, `P52W` or `P365D`. Defaults to `PT0S`.
 * `week_of_year` - (Optional) The week of year to take the yearly backup. Value has to be between `1` and `52`.
-* `immutable_backups_enabled` - (Optional) Specifies if the backups are immutable. Defaults to `false`.
 
 ---
 

